framework_tvb/tvb/adapters/datatypes/db/connectivity.py

```diff
@@ -67,6 +67,12 @@ def fill_from_has_traits(self, datatype):
         self.undirected = datatype.undirected
         self.weights_min, self.weights_max, self.weights_mean = from_ndarray(datatype.weights)
         self.tract_lengths_min, self.tract_lengths_max, self.tract_lengths_mean = from_ndarray(datatype.tract_lengths)
+        self.weights_min = numpy.float64(self.weights_min)
+        self.weights_mean = numpy.float64(self.weights_mean)
+        self.weights_max = numpy.float64(self.weights_max)
+        self.tract_lengths_min = numpy.float64(self.tract_lengths_min)
+        self.tract_lengths_mean = numpy.float64(self.tract_lengths_mean)
+        self.tract_lengths_max = numpy.float64(self.tract_lengths_max)
```

Comment:
Have you considered including the cast inside the from_ndarray?
It seems that all these variables come from it.